File: 0057-insert-interval/0057-insert-interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        
        if(len(intervals)==0):
            return [newInterval]
        
        mx=newInterval[1]
        mn=newInterval[0]
        ans=[]
        added=False
        
        
        for interval in intervals:
            s,e=interval[0],interval[1]
            
            if mn<=e and mx>=s:
                mx=max(mx,e)
                mn=min(mn,s)
            elif added!=True and mx<s:
                ans.append([mn,mx])
                ans.append(interval)
                added=True
            else:
                ans.append(interval)
            
        if added!=True:
            ans.append([mn,mx])
            
        return ans
            
                
                
        

File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        
        List<int[]> ans = new ArrayList<>();
        
        List<int[]> newIntervals = new ArrayList<>();
        
        
        for(int[] interval : intervals){
            
            newIntervals.add(interval);
        }
        
        newIntervals.add(newInterval);
        
        Collections.sort(newIntervals,(a,b)->{
           return a[0] - b[0];
        });
        
        int s = newIntervals.get(0)[0], e = newIntervals.get(0)[1];
        
        for(int i = 1; i < newIntervals.size(); i++){
            
            int currS = newIntervals.get(i)[0];
            int currE = newIntervals.get(i)[1];
            
            
            if(e < currS){
                ans.add(new int[]{s,e});
                e = currE;
                s = currS;
            }else {
                e = Math.max(e,currE);
            }
        }
        ans.add(new int[]{s,e});
        
        int[][] result = new int[ans.size()][2];
        
        
        int index = 0;
        
        for(int[] temp : ans){
            result[index++] = temp;
        }
        return result;
        
    }
}